migration.rst

```diff
@@ -0,0 +1,415 @@
+    have to tweak for your legacy application. You can familiarize
+    yourself with the concepts by reading up on it in :doc:`Routing </routing>`.
+
+The legacy route loader has a similar functionality as the previous
+LegacyBridge, but it is a service that is registered inside Symfony's
+routing component::
+
+    public function load($resource, $type = null)
+    {
+        $collection = new RouteCollection();
+        $finder = new Finder();
+        $finder->files()->name('*.php');
+
+        /** @var SplFileInfo $legacyScriptFile */
+        foreach ($finder->in($this->webDir) as $legacyScriptFile) {
+            // This assumes all legacy files use ".php" as extension
+            $filename = basename($legacyScriptFile->getRelativePathname(), '.php');
+            $routeName = sprintf('app.legacy.%s', str_replace('/', '__', $filename));
+
+            $collection->add($routeName, new Route($legacyScriptFile->getRelativePathname(), [
+                '_controller' => 'App\Controller\LegacyController::loadLegacyScript',
+                'requestPath' => '/' . $legacyScriptFile->getRelativePathname(),
+                'legacyScript' => $legacyScriptFile->getPathname(),
+            ]));
+        }
+
+        return $collection;
+    }
+
+This only shows the ``load``-method of a custom Route Loader. You will also
+have to register the loader in your application's ``routing.yaml`` as
+described in the documentation for :doc:`Custom Route Loaders </routing/custom_route_loader>`.
+Depending on your configuration you might also have to tag the service with
+``routing.loader``. Afterwards you should be able to see all the legacy routes
+in your route configuration, e.g. when you call the ``debug:router``-command:
+
+.. code-block:: terminal
+
+    $ php bin/console debug:router
+
+In order to use these routes you will need to create a controller that handles
+these routes. You might have noticed the ``_controller`` attribute we attached
+to our routes, which tells Symfony which Controller to call whenever we try
+to access one of our legacy routes. The controller itself can then use the
+attributes we passed to determine which script to call and wrap the output in
+a response class::
+
+    class LegacyController
+    {
+        public function loadLegacyScript($requestPath, $legacyScript)
+        {
+            return StreamedResponse::create(
+                function () use ($requestPath, $legacyScript) {
+                    $_SERVER['PHP_SELF'] = $requestPath;
+                    $_SERVER['SCRIPT_NAME'] = $requestPath;
+                    $_SERVER['SCRIPT_FILENAME'] = $legacyScript;
+
+                    chdir(dirname($legacyScript));
+
+                    require $legacyScript;
+                }
+            );
+        }
+    }
+
+This controller will set some server variables that might be needed by
+the legacy application, switches to the script directory to ensure all
```

Comment:
I find this phrase a bit hard to understand unless you read it carefully: `ensure all
operations assuming this is the current working dir work properly`  Maybe we can reword it? Also, we should replace `dir` by `directory` to be consistent with the rest of docs. Thanks!

Reply:
I have rephrased this section to make it a bit clearer